Xamarin.Essentials/Preferences/Preferences.win32.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Preferences
+    {
+        static Dictionary<string, object> preferences = new Dictionary<string, object>();
```

Comment:
Is this not actually saved to disk at all?

Reply:
Need some discussion over the implementation of this. This is temporary and required for the code to build and allow the sample app to run. There isn't a native API we can use (unless taking a dependency on the Windows 10 UWP API). A simple implementation would be to persist the dictionary(s) to a file but that may require a dependency on Newtonsoft.Json or similar. I think that will be the easiest option for now and we can discuss further the pros/cons.

Reply:
No need for newtonsoft, BinarySerializer etc. are all available and fit the scenario. Persistance is IMHO necessary for this to be considered workable.

Reply:
You could have a look here: https://docs.microsoft.com/en-us/dotnet/standard/serialization/

Reply:
Thanks, I'll rework this and use one of the built-in options.

Reply:
Going the file route comes with another set of problems:
We definitely have to have a lazy mechanism to store and retrieve settings. Loading all properties on first acess would increse the working set unnecessarily. Also, we do need fast key existance checks. Not only for Key enumeration but also for insertion. I am currently hooking up some different benchmarks for a couple of potential solutions, but this all seems very icky to me. Are there any usbale storage technologies available with winapi we could use? or even com perhaps

Reply:
Okay this could be a stupid idea but how about the registry? Fast key existance checks, key value store just as we like it and as persistent as it can be

Reply:
I think that's frowned upon, we can't control how much it gets used for and unless the app is installed via Centennial it'll get dumped into the main registry.